textalyze.js

```diff
@@ -14,10 +6,27 @@
 function itemCounts(array) {
   let counts = new Map();
 
-  // Your code here.
-  // Run 'npm test' to see what tests need to pass.
+  array.forEach(element => {
+    
+    if (!counts.has(element))
```

Comment:
Exercise: how would you write this as a single call to [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)?

Reply:
I would write this:
var counts = arr.reduce((accum, arrVal) => {
                var newCount = accum.has(arrVal) ? accum.get(arrVal) + 1 : 1;
                return accum.set(arrVal, newCount);
              }, new Map());